OSDOCS 8197: Added delete, dry run, and enclave support functionality for oc-mirror v2 (#77289)
openshift/openshift-docs: modules/oc-mirror-workflows-delete-v2.adoc

// Module included in the following assemblies:
//
// * installing/disconnected_install/installing-mirroring-disconnected-v2.adoc

> :_mod-docs-content-type: CONCEPT

Reviewer: Usually gerunds are reserved for proedures. How about:

> Deletion of images from your disconnected environment

In the following example, `DeleteImageSetConfiguration` removes the following:

* All images of {product-title} release 4.13.3.
* The catalog image `redhat-operator-index` `v4.12`.
* The `aws-load-balancer-operator` v0.0.1 bundle and all its related images.
* The additional images `ubi` and `ubi-minimal` referenced by their corresponding digests.

.Example: `DeleteImageSetConfig`
[source,yaml]
----
apiVersion: mirror.openshift.io/v2alpha1
kind: DeleteImageSetConfiguration
delete:
  platform:
    channels:
      - name: stable-4.13 
        minVersion: 4.13.3
        maxVersion: 4.13.3
  operators:
    - catalog: registry.redhat.io/redhat/redhat-operator-index:v4.12
      packages:
      - name: aws-load-balancer-operator
         minVersion: 0.0.1
         maxVersion: 0.0.1
  additionalImages: 
    - name: registry.redhat.io/ubi8/ubi@sha256:bce7e9f69fb7d4533447232478fd825811c760288f87a35699f9c8f030f2c1a6
    - name: registry.redhat.io/ubi8/ubi-minimal@sha256:8bedbe742f140108897fb3532068e8316900d9814f399d676ac78b46e740e34e
----

[IMPORTANT]
====

> It is highly recommended to use the mirror-to-disk and disk-to-mirror workflows.

Reviewer: ```suggestion
Consider using the mirror-to-disk and disk-to-mirror workflows to recduce mirroring issues.
```

> Consider using the mirror-to-disk and disk-to-mirror workflows to recduce mirroring issues.

Reviewer 2: ```suggestion
Consider using the mirror-to-disk and disk-to-mirror workflows to reduce mirroring issues.
```

====

In the image delete workflow, oc-mirror plugin v2 deletes only the manifests of the images, which does not reduce the storage occupied in the registry.

> To free up storage space from unnecessary images (those with deleted manifests), you must enable the garbage collector on your container registry. With the garbage collector enabled, the registry will delete the image blobs that no longer have references to any manifests, thereby reducing the storage previously occupied by the deleted blobs. Enabling the garbage collector differs depending on your container registry. 

Reviewer: ```suggestion
To free up storage space from unnecessary images, such as those with deleted manifests, you must enable the garbage collector on your container registry. With the garbage collector enabled, the registry will delete the image blobs that no longer have references to any manifests, thereby reducing the storage previously occupied by the deleted blobs. Enabling the garbage collector differs depending on your container registry. 
```
